File: app/src/main/java/org/kindone/willingtodo/pomodorocontrol/ControlInitial.java
package org.kindone.willingtodo.pomodorocontrol;

/**
 * Created by kindone on 2017. 1. 2..
 */

public class ControlInitial extends ControlState {
    public ControlInitial(ControlContext context) {
        super(context);
    }

    @Override
    public void start() {
        changeState(new ControlRunning(mContext));
    }

    @Override
    public void pause() {
        // DO_NOTHING
    }

    @Override
    public void resume() {
        // DO_NOTHING
    }

    @Override
    public void stop() {
        // DO_NOTHING
    }

    @Override
    public int getMode() {
        return MODE_PLAYING;
    }
}

File: app/src/main/java/org/kindone/willingtodo/pomodorocontrol/ControlInitial.kt
package org.kindone.willingtodo.pomodorocontrol

/**
 * Created by kindone on 2017. 1. 2..
 */

class ControlInitial(context: ControlContext) : ControlState(context) {

    override fun start() {
        changeState(ControlRunning(mContext))
    }

    override fun pause() {
        // DO_NOTHING
    }

    override fun resume() {
        // DO_NOTHING
    }

    override fun stop() {
        // DO_NOTHING
    }

    override fun getMode(): Int {
        return ControlState.MODE_PLAYING
    }
}
